### PR TITLE
fix: prepare for OTP 27

### PR DESCRIPTION
**Pull request by kivra-pergus**
Quotes definition and use of the maybe/2 type, also exports the type from s2_maybe because it was already used as if exported from there.

### CODE REVIEW

File: include/maybe.hrl [hunk 1]
```diff
@@ -10,7 +10,7 @@
 
 -type ok(A)       :: {ok, A}.
 -type error(A)    :: {error, A}.
--type maybe(A, B) :: ok(A) | error(B).
+-type 'maybe'(A, B) :: ok(A) | error(B).
```
Comment (kivra-fabber) [inline include/maybe.hrl:13]: **Super unimportant tangent**: If this module and type had been given the correct name (`result`) from the start we would have to resort to these apostrophes. But there is of course no guarantee `result` would not be reserved in another universe, or `either` which would also have been a sense-making name for this type.

The original stdlib2 authors chose to name the [Result type](https://en.wikipedia.org/wiki/Result_type) "maybe" for some reason. `Maybe` is the Haskell name of the [Option type](https://en.wikipedia.org/wiki/Option_type). Haskell of course has a result type as well, but it is called `Either`. So this `s2_maybe` business is very confusing.

Anywho, I don't see us renaming it here "just because" it would fix the apostrophe inconvenience...

File: src/s2_pn_counters.erl [hunk 2]
```diff
@@ -34,7 +34,7 @@
 %%%_* Code =============================================================
 %%%_ * Types -----------------------------------------------------------
 -define(M                 ,  s2_vclocks).
--opaque pn_counter()      :: {?M:new(), ?M:new()}.
+-opaque pn_counter()      :: {?M:vclock(), ?M:vclock()}.
```
Comment (kivra-fabber) [inline src/s2_pn_counters.erl:37]: Why this change in this PR?

File: src/stdlib2.app.src [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {application, stdlib2, [ {description,  "stdlib2"}
                        , {vsn,          git}
                        , {registered,   []}
-                       , {applications, [kernel, stdlib]}
+                       , {applications, [kernel, stdlib, compiler]}
```
Comment (kivra-fabber) [inline src/stdlib2.app.src:4]: What is the implication of this change for any library user? They will all get the compiler application now in their releases?

**Review by kivra-fabber — APPROVED** [review]
Looks good to me, take my questions/comments however you like.

File: src/s2_pn_counters.erl (hunk 2 above)
Comment (kivra-pergus) [reply src/s2_pn_counters.erl:37]: In order to make the CI work for all OTP versions, dialyzer would complain for OTP26 and OTP27 because of the unknown type. I'll change the description a bit to emphasise getting the library to work on later OTP versions.

**Comment by kivra-fabber** [comment]
What a cursed branch name!

```
❯ git checkout "'maybe'_notmaybe"
branch ''maybe'_notmaybe' set up to track 'origin/'maybe'_notmaybe'.
Switched to a new branch ''maybe'_notmaybe'
```

File: src/stdlib2.app.src (hunk 3 above)
Comment (kivra-pergus) [reply src/stdlib2.app.src:4]: Yes, but stdlib depends on compiler so they will already have it, but stdlib2 depends directly on compiler so this is dependency should be there. It is also necessary for dialyzer to pass.

File: include/maybe.hrl (hunk 1 above)
Comment (kivra-pergus) [reply include/maybe.hrl:13]: Yeah, the naming is what it is, but making a name change that is non-backwards compatible seems worse than making a backwards compatible (but ugly) change
